dm/loader/lightning.go

```diff
@@ -197,14 +216,20 @@ func (l *LightningLoader) restore(ctx context.Context) error {
 		cpPath := filepath.Join(l.cfg.LoaderConfig.Dir, lightningCheckpointFileName)
 		cfg.Checkpoint.DSN = cpPath
 		cfg.Checkpoint.KeepAfterSuccess = lcfg.CheckpointOrigin
+		cfg.TikvImporter.OnDuplicate = string(l.cfg.OnDuplicate)
 		cfg.TiDB.Vars = make(map[string]string)
+		cfg.Routes = l.cfg.RouteRules
 		if l.cfg.To.Session != nil {
 			for k, v := range l.cfg.To.Session {
 				cfg.TiDB.Vars[k] = v
 			}
 		}
-		cfg.TiDB.StrSQLMode = l.cfg.LoaderConfig.SQLMode
-		cfg.TiDB.Vars = map[string]string{"time_zone": l.timeZone}
+		cfg.TiDB.StrSQLMode = l.sqlMode
+		cfg.TiDB.Vars = map[string]string{
```

Comment:
how about w init `cfg.TiDB.Vars`  with tidb_txn_mode in `makeGlobalConfig` ?

Reply:
GloalConfig doesn't contains `Vars` field.